hack/generator/pkg/astmodel/storage_conversion_factories.go

```diff
@@ -342,36 +263,40 @@ func assignArrayFromArray(
 		return nil
 	}
 
-	return func(reader dst.Expr, writer dst.Expr, generationContext *CodeGenerationContext) []dst.Stmt {
+	return func(reader dst.Expr, writer func(dst.Expr) []dst.Stmt, generationContext *CodeGenerationContext) []dst.Stmt {
 		// We create three obviously related identifiers to use for the conversion
-		id := sourceEndpoint.CreateLocal()
-		itemId := id + "Item"
-		indexId := id + "Index"
-		tempId := id + "List"
+		itemId := sourceEndpoint.CreateLocal("Item")
+		indexId := sourceEndpoint.CreateLocal("Index")
+		tempId := sourceEndpoint.CreateLocal("List")
 
 		declaration := astbuilder.SimpleAssignment(
 			dst.NewIdent(tempId),
 			token.DEFINE,
 			astbuilder.MakeList(dstArray.AsType(generationContext), astbuilder.CallFunc("len", reader)))
 
-		body := conversion(
-			dst.NewIdent(itemId),
-			&dst.IndexExpr{
-				X:     dst.NewIdent(tempId),
-				Index: dst.NewIdent(indexId),
-			},
-			generationContext,
-		)
+		writeToElement := func(expr dst.Expr) []dst.Stmt {
+			return []dst.Stmt{
+				astbuilder.SimpleAssignment(
+					&dst.IndexExpr{
+						X:     dst.NewIdent(tempId),
+						Index: dst.NewIdent(indexId),
+					},
+					token.ASSIGN,
+					expr),
+			}
+		}
 
-		assign := astbuilder.SimpleAssignment(writer, token.ASSIGN, dst.NewIdent(tempId))
+		avoidAliasing := astbuilder.SimpleAssignment(dst.NewIdent(itemId), token.DEFINE, dst.NewIdent(itemId))
```

Comment:
Add this to the example code in the doc comment?

Reply:
Done.